src/v/cluster/controller_backend.h

```diff
@@ -364,6 +364,13 @@ class controller_backend
       const std::vector<model::broker_shard>&,
       model::revision_id);
 
+    ss::future<std::error_code> force_replica_set_update(
+      const model::ntp&,
+      const std::vector<model::broker_shard>&,
+      const replicas_revision_map&,
+      const std::vector<model::broker_shard>&,
```

Comment:
nit: parameter names would be nice, it is difficult to distinguish `const std::vector<model::broker_shard>&` from `const std::vector<model::broker_shard>&` :)